src/util/orders.ts

```diff
@@ -35,3 +38,52 @@ export const buildOrder = (params: BuildOrderParams, side: OrderSide): Order =>
         senderAddress: '0x0000000000000000000000000000000000000000',
     };
 };
+
+export const orderDetailsFeeEther = (
+    makerAmount: BigNumber,
+    takerAmount: BigNumber,
+    orderType: OrderSide,
+    tokenDecimals: number = 18,
+): BigNumber => {
+    let totalFee = new BigNumber(1);
+
+    // Convert the makerAmount
+    const makerAmountConverted = new BigNumber(tokenAmountInUnits(makerAmount, tokenDecimals));
+    if (orderType === OrderSide.Buy) {
+        // Calculate makerFee
+        const makerFeeWithoutPrice = makerAmountConverted.mul(MAKER_FEE);
+        const makerFeeWithPrice = makerFeeWithoutPrice.mul(takerAmount.div(makerAmountConverted));
```

Comment:
Is this correct? When you create a buy order, the maker asset is wETH, so it shouldn't be multiplied by the price.

Reply:
The conditional logic was modified. Can you check it ? Thanks

Reply:
Now it _looks_ correct, but one of the tests seems wrong.

Reply:
Hey guys, sorry to jump-in late in the thread.

I believe we just have to convert the `makerFee` value (wich is always in ZRX) to ETH, without involving the order's asset amounts, price, and/or the `takerFee`.

So... the way I see things, the *Total Cost* showed to the user should only involve the `makerFee`: i.e. as a user, I'm mostly interested in "how much _I have to pay_". The `takerFee` will be the taker's thing to deal with.

I'm not 100% sure of this, we can confirm this with the 0x team tomorrow.